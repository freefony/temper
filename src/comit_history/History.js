import React, { useState } from 'react'

import { TimeTravelBtn } from './TimeTravelBtn'

export default ({title, from, to, position, onTimeTravel}) => {
  const [animation, setAnimation] = useState('slideInLeft')
  const toggleState = (position) => {
    const slideWay = (animation === 'slideInLeft') ? 'slideOutRight': 'slideInLeft'
    setAnimation(slideWay)
    setTimeout(() => onTimeTravel(position), 1000)
  }
  console.log(animation)
  return <div className={`px-2 py-4 flex flex-no-wrap justify-between bg-white border text-sm animated ${animation} fast`}>
<p className="text-gray-700 text-left">Moved {title} from index {from}  to index {to}</p>
    <span><TimeTravelBtn position={position} timeTravelAction={toggleState} /></span>
  </div>
}