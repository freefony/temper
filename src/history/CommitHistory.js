import React, { Component } from 'react'

class CommitHistory extends Component {
  render() {
    return <div className="container">
      <div className='text-2xl text-gray-600 text-left max-w-sm shadow-lg bg-white shadow overflow-hidden p-2'>List of actions commited</div>
      <div className="max-w-sm  bg-gray-100 shadow overflow-hidden p-4">
        <div className="px-6 py-4 flex justify-between bg-white border">
          <p className="text-gray-700 text-base">History</p>
          <span>></span>
        </div>
      </div>
    </div>
  }
}

export default CommitHistory