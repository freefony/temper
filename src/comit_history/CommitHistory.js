import React, { Component } from 'react'

import History from './History'

class CommitHistory extends Component {
  render() {
    const { history, onTimeTravel } = this.props

    
    return <div className="container">
      <div className='text-2xl text-gray-600 text-left max-w-sm shadow-lg bg-white shadow overflow-hidden p-2'>List of actions commited</div>
      <div className="max-w-sm  bg-gray-100 shadow overflow-hidden p-4">
      {history.map((historyObj, i) => 
        <History 
          title={historyObj.title} 
          from={historyObj.from} 
          to={historyObj.to} key={i} 
          position={i}
          onTimeTravel={onTimeTravel}/> 
      )}
      </div>
    </div>
  }
}

export default CommitHistory