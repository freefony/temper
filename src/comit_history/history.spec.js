import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import CommitHistory from './CommitHistory'
import History from './History'

import data from '../../fixtures/history'

describe('History', () => {
  afterEach(cleanup)
  describe('CommitHistory', () => {
    it('should render component', () => {
      const { container } = render(<CommitHistory />)
      expect(container).toMatchSnapshot()
    })
    it('should display history cards with correct text', () => {
      const { container, getByText } = render(<CommitHistory history={data.history} />)
      expect(container).toMatchSnapshot()
      expect(getByText(/moved Post 1 from index 0 to index 1/i)).toBeInTheDocument()
      expect(getByText(/moved Post 1 from index 1 to index 2/i)).toBeInTheDocument()
      expect(getByText(/moved Post 2 from index 4 to index 3/i)).toBeInTheDocument()
      expect(getByText(/moved Post 3 from index 1 to index 0/i)).toBeInTheDocument()
    })
  })
  describe('History', () => {
    const mockTimeTravelFnc = jest.fn(() => 1)
    it('should display text and time travel btn', () => {
      const { container, getByText } = render(<History {...data.history[0]} position={0} onTimeTravel={mockTimeTravelFnc} />)
      expect(container).toMatchSnapshot()
      expect(getByText(/moved Post 1 from index 0 to index 1/i)).toBeInTheDocument()
      
    })
    it('should contain a clickable btn that calls the timeTravelAction fnc props when clicked', () => {
      const { container, queryByText } = render(<History 
          {...data.history[0]} 
          position={0} 
          onTimeTravel={mockTimeTravelFnc} />)
      expect(container).toMatchSnapshot()
      expect(queryByText('Time travel')).toBeInTheDocument()
      fireEvent(
        queryByText('Time travel'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      )
      expect(mockTimeTravelFnc).toHaveBeenCalledWith(0)
    })
  })
})