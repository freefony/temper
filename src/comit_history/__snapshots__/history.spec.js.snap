// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History CommitHistory should display history cards with correct text 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="text-2xl text-gray-600 text-left max-w-sm shadow-xl bg-white shadow overflow-hidden p-2 flex-1"
    >
      List of actions commited
    </div>
    <div
      class="max-w-sm  bg-gray-100 shadow-lg overflow-hidden p-4"
    >
      <div>
        <div
          class="px-2 py-4 flex flex-no-wrap justify-between bg-white border text-sm"
        >
          <p
            class="text-gray-700 text-left"
          >
            Moved 
            Post 3
             from index 
            1
              to index 
            0
          </p>
          <div
            class="flex-none"
          >
            <button
              class="bg-green-300 hover:bg-green-200 text-gray-700 font-bold py-2 px-2 rounded"
            >
              Time travel
            </button>
          </div>
        </div>
        <div
          class="px-2 py-4 flex flex-no-wrap justify-between bg-white border text-sm"
        >
          <p
            class="text-gray-700 text-left"
          >
            Moved 
            Post 2
             from index 
            4
              to index 
            3
          </p>
          <div
            class="flex-none"
          >
            <button
              class="bg-green-300 hover:bg-green-200 text-gray-700 font-bold py-2 px-2 rounded"
            >
              Time travel
            </button>
          </div>
        </div>
        <div
          class="px-2 py-4 flex flex-no-wrap justify-between bg-white border text-sm"
        >
          <p
            class="text-gray-700 text-left"
          >
            Moved 
            Post 1
             from index 
            1
              to index 
            2
          </p>
          <div
            class="flex-none"
          >
            <button
              class="bg-green-300 hover:bg-green-200 text-gray-700 font-bold py-2 px-2 rounded"
            >
              Time travel
            </button>
          </div>
        </div>
        <div
          class="px-2 py-4 flex flex-no-wrap justify-between bg-white border text-sm"
        >
          <p
            class="text-gray-700 text-left"
          >
            Moved 
            Post 1
             from index 
            0
              to index 
            1
          </p>
          <div
            class="flex-none"
          >
            <button
              class="bg-green-300 hover:bg-green-200 text-gray-700 font-bold py-2 px-2 rounded"
            >
              Time travel
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`History CommitHistory should render component 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="text-2xl text-gray-600 text-left max-w-sm shadow-xl bg-white shadow overflow-hidden p-2 flex-1"
    >
      List of actions commited
    </div>
    <div
      class="max-w-sm  bg-gray-100 shadow-lg overflow-hidden p-4"
    >
      <div />
    </div>
  </div>
</div>
`;

exports[`History History should contain a clickable btn that calls the timeTravelAction fnc props when clicked 1`] = `
<div>
  <div
    class="px-2 py-4 flex flex-no-wrap justify-between bg-white border text-sm"
  >
    <p
      class="text-gray-700 text-left"
    >
      Moved 
      Post 1
       from index 
      0
        to index 
      1
    </p>
    <div
      class="flex-none"
    >
      <button
        class="bg-green-300 hover:bg-green-200 text-gray-700 font-bold py-2 px-2 rounded"
      >
        Time travel
      </button>
    </div>
  </div>
</div>
`;

exports[`History History should display text and time travel btn 1`] = `
<div>
  <div
    class="px-2 py-4 flex flex-no-wrap justify-between bg-white border text-sm"
  >
    <p
      class="text-gray-700 text-left"
    >
      Moved 
      Post 1
       from index 
      0
        to index 
      1
    </p>
    <div
      class="flex-none"
    >
      <button
        class="bg-green-300 hover:bg-green-200 text-gray-700 font-bold py-2 px-2 rounded"
      >
        Time travel
      </button>
    </div>
  </div>
</div>
`;
